istioctl/cmd/analyze.go

```diff
@@ -172,13 +193,30 @@ istioctl experimental analyze -k -d false
 				}
 			}
 
-			// Print validation message output, or a line indicating that none were found
-			if len(outputMessages) == 0 {
-				fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
-			} else {
-				for _, m := range outputMessages {
-					fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
+			switch msgOutputFormat {
+			case LogOutput:
+				// Print validation message output, or a line indicating that none were found
+				if len(outputMessages) == 0 {
+					fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
+				} else {
+					for _, m := range outputMessages {
+						fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
+					}
+				}
+			case JsonOutput:
+				jsonOutput, err := json.Marshal(outputMessages)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintln(cmd.OutOrStdout(), string(jsonOutput))
+			case YamlOutput:
+				yamlOutput, err := yaml.Marshal(outputMessages)
+				if err != nil {
+					return err
 				}
+				fmt.Fprintln(cmd.OutOrStdout(), string(yamlOutput))
+			default: // This should never happen since we validate this already
+				return fmt.Errorf("%q not found in output format switch statement post validate?", msgOutputFormat)
```

Comment:
If we keep this as a "never should possibly happen" case then just panic. I'm also OK simply deleting it if the linter will let us.